### PR TITLE
Add POSIX support

### PR DESCRIPTION
**Pull request by pranjalchanda08**

### CODE REVIEW

**Comment by pranjalchanda08** [comment]
> #define EPERM error_op_not_perm // This is from status.h

We need to redefine all the codes. If we see all are positive integers. But CC works with negative error codes.

**Comment by akashkollipara** [comment]
> > #define EPERM error_op_not_perm // This is from status.h
> 
> We need to redefine all the codes. If we see all are positive integers. But CC works with negative error codes.

Yes... but use double -ve
like this `#define ERERM -error_op_not_perm`

File: src/lib/posix/include/time.h [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+ * @brief Report CPU time used.
+ *
+ * http://pubs.opengroup.org/onlinepubs/9699919799/functions/clock.html
+ *
+ * @return  The number of Kernel ticks since the scheduler
+ * was started minus the ticks spent in the idle task.
+ *
+ * @note This function does NOT report the number of ticks spent by the calling thread.
+ */
+clock_t clock( void );
```
Comment (akashkollipara) [inline src/lib/posix/include/time.h:82]: for this we shall use our custom libc time.h format.

File: src/lib/posix/cc_posix_internal.h [hunk 2]
```diff
@@ -0,0 +1,100 @@
+#ifndef _CC_POSIX_INTERNAL_H_
```
Comment (akashkollipara) [inline src/lib/posix/cc_posix_internal.h:1]: replace this with `#pragma once`

**Comment by akashkollipara** [comment]
bro this branch has hung amount merge conflict with development... which is getting tracked indirectly
please have a look

**Comment by akashkollipara** [comment]
Fixed the merge conflicts, but still the history is not linear...

**Comment by akashkollipara** [comment]
Merging it into cc-kernel to avoid conflicts...

**Pull request merged**